import-automation/executor/app/executor/import_executor.py

```diff
@@ -813,15 +812,22 @@ def _update_import_status_table(
 
     def _update_latest_version(self, version, output_dir, import_spec,
                                import_summary):
+        if self.config.skip_gcs_upload:
+            logging.warning(
+                "Skipping latest version update as per import config.")
+            return
         logging.info(f'Updating import latest version {version}')
         self.uploader.upload_string(
             version,
             os.path.join(output_dir, self.config.storage_version_filename))
         self.uploader.upload_string(
             self._import_metadata_mcf_helper(import_spec),
             os.path.join(output_dir, self.config.import_metadata_mcf_filename))
-        if self.config.storage_version_history_filename:
-            # Add current version to the history of versions.
+        self.uploader.upload_string(
+            json.dumps(dataclasses.asdict(import_summary), default=str),
+            os.path.join(output_dir, 'import_summary.txt'))
+        # Add current version to the history of versions if import was successful.
+        if self.config.storage_version_history_filename and import_summary.status == ImportStatus.READY:
             history_filename = os.path.join(
                 output_dir, self.config.storage_version_history_filename)
```

Comment:
version_history should be updated if latest_version is updated (line 820) even for overrides.
should we have the successful check for latest_version.txt as well?